File: screenshot/ViewControllers/ScreenshotDisplay/ScreenshotDisplayNavigationController.m
//
//  ScreenshotDisplayNavigationController.m
//  screenshot
//
//  Created by Corey Werner on 8/22/17.
//  Copyright © 2017 crazeapp. All rights reserved.
//

#import "ScreenshotDisplayNavigationController.h"

@interface ScreenshotDisplayNavigationController ()

@end

@implementation ScreenshotDisplayNavigationController

#pragma mark - Life Cycle

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        [self.navigationBar setBackgroundImage:[[UIImage alloc] init] forBarMetrics:UIBarMetricsDefault];
        self.navigationBar.shadowImage = [[UIImage alloc] init];
        self.navigationBar.translucent = YES;
        self.navigationBar.tintColor = [UIColor whiteColor];
        
        _screenshotDisplayViewController = ({
            UIImage *image = [[UIImage imageNamed:@"LogoHollowC"] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
            
            UIImageView *imageView = [[UIImageView alloc] initWithImage:image];
            imageView.tintColor = [UIColor whiteColor];
            
            ScreenshotDisplayViewController *viewController = [[ScreenshotDisplayViewController alloc] init];
            viewController.navigationItem.titleView = imageView;
            viewController.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"ControlX"] style:UIBarButtonItemStylePlain target:self action:@selector(closeAction)];
            viewController.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemAction target:self action:@selector(shareAction)];
            viewController;
        });
        
        self.viewControllers = @[self.screenshotDisplayViewController];
    }
    return self;
}


#pragma mark - Actions

- (void)closeAction {
    [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

- (void)shareAction {
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Screenshot Sharing" message:@"Coming soon." preferredStyle:UIAlertControllerStyleAlert];
    [alertController addAction:[UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleDefault handler:nil]];
    [self presentViewController:alertController animated:YES completion:nil];
}

@end

File: screenshot/View/ScreenshotDisplay/ScreenshotDisplayNavigationController.swift
//
//  ScreenshotDisplayNavigationController.swift
//  screenshot
//
//  Created by Corey Werner on 1/3/18.
//  Copyright (c) 2018 crazeapp. All rights reserved.
//

import Foundation
import UIKit

class ScreenshotDisplayNavigationController : UINavigationController {
    let screenshotDisplayViewController = ScreenshotDisplayViewController()
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    convenience init() {
        self.init(nibName: nil, bundle: nil)
    }
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
        
        navigationBar.setBackgroundImage(UIImage(), for: .default)
        navigationBar.shadowImage = UIImage()
        navigationBar.isTranslucent = true
        navigationBar.tintColor = .white
        
        screenshotDisplayViewController.navigationItem.titleView = UIImageView(image: UIImage(named: "BrandLogoWhite20h"))
        screenshotDisplayViewController.navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(named: "ControlX"), style: .plain, target: self, action: #selector(closeAction))
//        screenshotDisplayViewController.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .action, target: self, action: #selector(shareAction))
        
        viewControllers = [screenshotDisplayViewController]
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    
    @objc fileprivate func closeAction() {
        presentingViewController?.dismiss(animated: true, completion: nil)
    }
    
    @objc fileprivate func shareAction() {
        // TODO:
    }
}
